driver/src/main/java/eu/cloudnetservice/cloudnet/driver/permission/PermissionUser.java

```diff
@@ -76,41 +109,105 @@ protected PermissionUser(
       .groupPermissions(user.groupPermissions());
   }
 
+  /**
+   * Checks if the given password matches the stored password. The input is hashed using sha 256 and encoded using
+   * base64 and then compared to the stored value.
+   *
+   * @param password the password to compare.
+   * @return true if the password matches, false otherwise.
+   */
   public boolean checkPassword(@Nullable String password) {
     return this.hashedPassword != null
       && password != null
       && this.hashedPassword.equals(Base64.getEncoder().encodeToString(HashUtil.toSha256(password)));
   }
 
+  /**
+   * Gets the unique id of the permission user.
+   *
+   * @return the unique id of the user.
+   */
   public @NonNull UUID uniqueId() {
     return this.uniqueId;
   }
 
+  /**
+   * Gets all groups that the user belongs to.
+   *
+   * @return all groups of the user.
+   */
   public @NonNull Collection<PermissionUserGroupInfo> groups() {
     return this.groups;
   }
 
+  /**
+   * Gets the password of the user hashed with sha 256. If no password is set null is returned.
+   *
+   * @return the hashed password.
+   */
   public @Nullable String hashedPassword() {
     return this.hashedPassword;
   }
 
+  /**
+   * Gets the name of all groups the user is in.
+   *
+   * @return the name of all groups of the user.
+   */
   @Override
   public @NonNull Collection<String> groupNames() {
```

Comment:
Can you recheck if this collection must be modifyable? If not we can use .toList here